src/app_engine/compute_page.py

```diff
@@ -50,9 +63,30 @@ def _build_compute_service(self):
                          service_name='compute',
                          version='v1')
 
-  def _enqueue_start_task(self, instance, zone):
-    taskqueue.add(url='/compute/start/%s/%s' % (instance, zone),
-                  countdown=TASK_RETRY_S)
+  def _maybe_restart_instance(self, instance, zone):
+    """Implementation for restart action.
+
+    Args:
+      instance: Name of the instance to start.
+      zone: Name of the zone the instance belongs to.
+
+    """
+    if self.compute_service is None:
+      logging.warning('Compute service unavailable.')
+      return
+
+    status = self._compute_status(instance, zone)
+
+    logging.info('GCE VM \'%s (%s)\' status: \'%s\'.',
+                 instance, zone, status)
+
+    if status == COMPUTE_STATUS_RUNNING:
```

Comment:
What if it is in one of the other states mentioned in the start method?
      # Instance is in an intermediate state: PROVISIONING, STAGING,
      # STOPPING. Thus, requeue the task.
Can we get into a state where we start an extra instead of restarting?